centreon-certified/prometheus/prometheus-pushgateway-metrics-apiv2.lua

```diff
@@ -0,0 +1,613 @@
+    sdesc = self.sc_event.event.service_description
+  end
+  local hname = self.sc_event.event.cache.host.name
+
+  if (unit ~= '') then
+    if (self.enable_extended_metric_name == 0) then
+      name = label .. '_' .. unit
+    else
+      name = hname .. '_' .. sdesc .. ':' .. label .. '_' .. unit
+    end
+  else
+    if (self.enable_extended_metric_name == 0) then
+      name = label
+    else
+      name = hname .. '_' .. sdesc .. ':' .. label
+    end
+  end
+
+  return convert_to_openmetric(name)
+end
+
+--------------------------------------------------------------------------------
+---- EventQueue:format_metric_service method
+-- @param metric {table} a single metric data
+-------------------------------------------------------------------------------
+function EventQueue:format_metric_event(metric)
+  self.sc_logger:debug("[EventQueue:format_metric]: start real format metric ")
+  local event = self.sc_event.event
+  local type  = self:get_metric_type(metric)
+  local unit  = unit_mapping(metric.uom)
+  local label = metric.metric_name
+  local name  = self:create_metric_name(label, unit)
+  local data = ''
+  local sdesc = event.formated_event.prom_sdesc
+
+  data = '# TYPE ' .. name .. ' ' .. type .. '\n'
+  data = data .. self:add_unit_info(label, unit, name)
+  
+  if not event.hostgroupsLabel then
+    data = data .. name .. '{label="' .. label .. '", host="' .. event.cache.host.name .. '", service="' .. sdesc .. '"} ' .. metric.value .. '\n'
+  else
+    data = data .. name .. '{label="' .. label .. '", host="' .. event.cache.host.name .. '", service="' .. sdesc .. '", ' ..  event.hostgroupsLabel .. '} ' .. metric.value .. '\n'
+  end
+
+  if (self.enable_threshold_metrics == 1) then 
+    data = data .. self:threshold_metrics(metric, label, unit, type)
+  end
+
+  event.formated_event.payload = data
+
+  self:add()
+  self.sc_logger:debug("[EventQueue:format_metric]: end real format metric ")
+end
+
+--------------------------------------------------------------------------------
+-- ifnumber_not_nan: [for Prometheus] check if a number is a number (and not a NaN)
+-- @param {number} number, the number to check
+-- @return {boolean}
+--------------------------------------------------------------------------------
+local function ifnumber_not_nan (number)
+  if (number ~= number) then
+    return false
+  elseif (type(number) ~= 'number') then
+    return false
+  else
+    return true
+  end
+end
+
+--------------------------------------------------------------------------------
+-- get_metric_type: [for Prometheus] find out the metric type to match openmetrics standard
+-- @param {table} perfdata, the perfdata informations
+-- @return {string} metricType, the type of the metric
+--------------------------------------------------------------------------------
+function EventQueue:get_metric_type (perfdata)
+  local metricType = nil;
+  if (ifnumber_not_nan(perfdata.max)) then
```

Comment:
will need to change this function call to use the proper one

Reply:
done here https://github.com/centreon/centreon-stream-connector-scripts/pull/279/commits/76eab9f8f000eb1958378c06306ed5ae4f572c41